src/cryptonote_core/service_node_list.cpp

```diff
@@ -2004,16 +2105,19 @@ bool service_node_list::state_t::process_confirmed_event(
                     key,
                     confirm.confirmed_height);
         insert_info(key, std::move(service_node_info));
-        return true;
+        result.success = true;
+        result.need_swarm_update = true;
+        return result;
     } catch (const std::exception& e) {
         log::error(logcat, "Failed to register node from ethereum transaction: {}", e.what());
-        return false;
+        return result;
     }
 }
```

Comment:
A tiny cleanup here would be to delete the two `return result;`s above (in the try, and in catch) and add a single `return result;` at the bottom of the function.